src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -34,64 +35,165 @@ namespace NodeStore {
 
 constexpr std::uint32_t DatabaseShard::ledgersPerShardDefault;
 
-DatabaseShardImp::DatabaseShardImp(Application& app,
-    std::string const& name, Stoppable& parent, Scheduler& scheduler,
-    int readThreads, Section const& config, beast::Journal j)
-    : DatabaseShard(name, parent, scheduler, readThreads, config, j)
+DatabaseShardImp::DatabaseShardImp(
+    Application& app,
+    std::string const& name,
+    Stoppable& parent,
+    Scheduler& scheduler,
+    int readThreads,
+    beast::Journal j)
+    : DatabaseShard(
+        name,
+        parent,
+        scheduler,
+        readThreads,
+        app.config().section(ConfigSection::shardDatabase()),
+        j)
     , app_(app)
-    , config_(config)
-    , dir_(get<std::string>(config, "path"))
+    , ctx_(std::make_unique<nudb::context>())
+    , config_(app.config().section(ConfigSection::shardDatabase()))
+    , dir_(get<std::string>(config_, "path"))
     , backendName_(Manager::instance().find(
-        get<std::string>(config_, "type"))->getName())
-    , maxDiskSpace_(get<std::uint64_t>(config, "max_size_gb") << 30)
+        get<std::string>(config_, "type", "nudb"))->getName())
```

Comment:
Could `Manager::instance().find` return a nullptr (and crash on accessing `getName`)? We could just insert the string directly from the config and let `init` handle the error.

Reply:
Nice find!